### ISSUE TITLE
Load plugins from per-machine directory

### ISSUE THREAD

**Issue by cblichmann** [Vector35/binaryninja-api#1615]
There is no default path for Binary Ninja, since it is usually installed per-user (i.e. a directory in the users' home dir).

For tools like BinDiff, that ship with an installer/package that installs per-machine, this makes it necessary to ask users during installation where a particular tool can be found.

We could simply place plugin modules like `libbinexport11.so` (built from [here](https://github.com/cblichmann/binexport/blob/v11-binja)) into `~/.binaryninja/plugins`.
However, in Debian install scripts, for example, there is no clean way to get the installing user (it'll be `root`). On Windows and macOS this is easier, but it's still somewhat inelegant to have a system-wide installer dropping files into per-user folder.

TL;DR - Please consider loading plugins from additional well-kown per-machine paths.
E.g.:
- **Linux** `/etc/opt/binaryninja/plugins`
- **macOS** `/Application Support/Vector35/BinaryNinja/plugins`
- **Windows** `%ProgramData%/Vector35/BinaryNinja/plugins`

Files in per-user directories should take precedence over per-machine ones.

**Comment by psifertex** [comment]
There is a work-around which is that inside the user folder (which is at a predictable location: https://docs.binary.ninja/getting-started.html#user-folder) there is a file called "lastrun" which contains the path to the install for just this problem. 

Of course there is a minor caveat that if the user has multiple installs in different locations, the last-one-run will be in the file, but that should be sufficient for most workflows.

**Comment by psifertex** [comment]
Oh, reading more closely I'm not sure this helps. I hadn't considered the Debian install script not even knowing the current user running. 🤔

**Comment by plafosse** [comment]
We're considering these paths:

- Linux: `/opt/binaryninja/plugins'
- MacOS: `/Library/Application Support/Binary Ninja/plugins`
- Windows: `%ProgramData%/Vector35/Binary Ninja/plugins`

**Comment by cblichmann** [comment]
Thanks, that'll work!

In my Linux suggestion I was assuming that Binja would load config files from there as well, hence the path suggestion (see [FHS spec](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch03s07.html#etcoptConfigurationFilesForOpt)). BinDiff would've just liked its plugins there.

**Comment by psifertex** [comment]
We've got separate plans that will allow for custom user folders but those will probably be environment variable based. We're looking at this solution just for the specific use-case of "global installer shouldn't have to know where Binja is or what users are running binja" but we could always expand that decision later.

Also, likely blocking on #1585 since we need to resolve the canonicalization issue with org name before we can decide on the Windows path in particular.
